### PR TITLE
chore: minor refactoring

### PR DESCRIPTION
**Pull request by XhmikosR**
This is based on #930, so only the second patch is relevant.

Opening for discussion and I can split the changes you want later.

~~1. added `types` is package.json~~
~~2. added rimraf because the build dir wasn't being cleaned and could keep old files~~
3. a couple of minor refactorings which are subjective, so let me know if you want those
~~4. disabled source maps in TypeScript since the maps aren't included in the package; if they are supposed to be used in tests, maybe there should be a different production and dev config?~~

**- Summary**

**- Test plan**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by XhmikosR** [comment]
I removed the `sourcemap: false` change from tsconfig, but I still think you should look at disabling sourcemap for production since you don't ship them in the npm package but they are still referenced. Not a huge deal, just saves a few bytes and it's proper IMHO :)

File: src/runtimes/node/bundlers/esbuild/plugin_native_modules.ts [hunk 1]
```diff
@@ -48,23 +48,20 @@ const getNativeModulesPlugin = (externalizedModules: NativeNodeModules): Plugin
...
-            if (externalizedModules[args.path] === undefined) {
-              // eslint-disable-next-line no-param-reassign
-              externalizedModules[args.path] = {}
-            }
-
-            // eslint-disable-next-line no-param-reassign
-            externalizedModules[args.path][modulePath] = packageJsonData.version
-
-            return { path: args.path, external: true }
+          if (isNative === false) {
```
Comment (XhmikosR) [inline src/runtimes/node/bundlers/esbuild/plugin_native_modules.ts:51]: Not 100% sure it's fine, so the more eyes the better. :)

**Comment by netlify-team-account-1** [comment]
Hey @XhmikosR! Thank you for your continued contributions to our projects, we really appreciate your help. We aim for PRs to fix a bug / introduce a new feature. For refactorings (like this PR), it's probably best to discuss them more holistically first, for example in an issue. Could you maybe open up an issue for this, so we can discuss potential changes to code style rules / linting behind this PR?

**Comment by XhmikosR** [comment]
This was part of another branch I have, I don't mind closing this since it's not worth discussing such a minor refactoring. :)

**Pull request closed without merging**